### PR TITLE
HZN-985 / HZN-987: add node location to alarm/notification UI

### PR DESCRIPTION
**Pull request by RangerRick**
This PR adds location to the alarm and notification UIs.  It also fixes up some issues with the event browser showing locations and adds support for showing both the node and event source locations, and contains a clarification in the outage UI.

Where appropriate, I always refer to a location associated with a node as a "Node Location".  Otherwise event locations are either just "Location" or "Event Source Location".

JIRA: 
* http://issues.opennms.org/browse/HZN-985
* http://issues.opennms.org/browse/HZN-987

Related JIRA issues:
* http://issues.opennms.org/browse/HZN-314
* http://issues.opennms.org/browse/HZN-315

### CODE REVIEW

File: opennms-webapp/src/main/java/org/opennms/web/event/filter/NegativeLocationFilter.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public String toString() {
     @Override
     public boolean equals(Object obj) {
         if (obj == null) return false;
-        if (!(obj instanceof NegativeLocationFilter)) return false;
+        if (!(obj instanceof NegativeNodeLocationFilter)) return false;
```
Comment (agalue) [inline opennms-webapp/src/main/java/org/opennms/web/event/filter/NegativeLocationFilter.java]: I think it should be NegativeLocationFilter (like the class name), as there is another class called NegativeNodeLocationFilter

Comment (RangerRick) [inline opennms-webapp/src/main/java/org/opennms/web/event/filter/NegativeLocationFilter.java]: Oop, good catch... cut&paste error.  There is so much code duplication in these damn filters.  :/

**Comment by agalue** [comment]
I tested this branch with my test environment, getting data locally, and through a minion. The filters seems to be working fine on events, alarms, notification and outages.

On the alarms page (full details), it is not clear the notion of "Location" and "Node Location" as there are no explicit columns for them like you can see on the events. Maybe add a small label would help.

**Comment by agalue** [comment]
The event detail page shows the location, but not the "Node Location".
The alarm detail page doesn't show the location and node location.
The notification detail page doesn't show the location.

**Comment by RangerRick** [comment]
As for the alarm detail page, I'm not really sure how to label it without taking up too much space, beyond what's in the headers.  Unless we want to completely rearrange that page, or start putting icons next to them or something.

As far as the detail pages, it is true, I did not even look at them, because it didn't occur to me to do so, since that wasn't what the issues said to do.  :)  I can definitely add that info tho'.

**Comment by RangerRick** [comment]
On second thought, maybe I should rearrange the alarm page to be like the event one and put the log message on a second row.  The rows wouldn't be any taller, they'd just be laid out better.

**Comment by agalue** [comment]
> On Jan 11, 2017, at 10:26 AM, Benjamin Reed <notifications@github.com> wrote:
> 
> On second thought, maybe I should rearrange the alarm page to be like the event one and put the log message on a second row. The rows wouldn't be any taller, they'd just be laid out better.

Sounds good to me

**Comment by RangerRick** [comment]
OK, the latest commits add info to the detail pages, as well as cleaning up the alarm list page and such.

**Comment by RangerRick** [comment]
Event List:
event-list

Event Detail:
event-detail

Alarm List:
alarm-list

Alarm Detail:
alarm-detail

Notification List:
notification-list

Notification Detail:
notification-detail

Outage Detail:
outage-detail

**Comment by agalue** [comment]
Looks good to me, ship it!

**Comment by RangerRick** [comment]
I'll wait for the latest bamboo build to finish and merge it.  Thanks for checking things out.

**Pull request merged**